doc/guides/dts/index.rst

```diff
@@ -398,6 +398,91 @@ usually take a ``struct device*`` as their first argument. This allows the
 driver API to use information from devicetree to interact with the device
 hardware.
 
+Generated macro rules
+=====================
+.. code-block:: none
+
+      aliases {
+           i2c-0 = &i2c0;
+      };
+
+      soc {
+           i2c0: i2c@40066000 {
+                compatible = "nxp,kinetis-i2c";
+                reg = <0x40066000 0x1000>;
+                fxos8700@1d {
+                     compatible = "nxp,fxos8700";
+                     reg = <0x1d>;
+                };
+           };
+      };
+
+.. list-table::
+   :header-rows: 1
+
+   * - type
+     - prefix
+     - description
+     - example
+
+   * - node
+     - ``DT_<IDENT>_...``
+     -
+     - ``DT_NXP_KINETIS_I2C_40066000_...``
+   * - node
+     - ``DT_INST_<n>_<COMPAT>_...``
+     -
+     - ``DT_INST_0_NXP_KINETIS_I2C_...``
+   * - node
+     - ``DT_ALIAS_<ALIAS>_...``
+     -
+     - ``DT_ALIAS_I2C_0_...``
+   * - node
+     - ``DT_<COMPAT>_<ALIAS>_...``
+     -
+     - ``DT_NXP_KINETIS_I2C_I2C_0_...``
+   * - existence
+     - ``#define DT_COMPAT_<COMPAT> 1``
```

Comment:
well, can we get rid of the `DT_COMPAT_<COMPAT>_` and `DT_<COMPAT>_BUS_` if we have `DT_INST_`?